### PR TITLE
Fix building with custom llvm versions

### PR DESCRIPTION
**Pull request by lzaoral**
This PR should fix builds with custom `llvm-config=`.

### CODE REVIEW

File: system-build.sh [hunk 1]
```diff
@@ -234,30 +234,20 @@ export LLVM_PREFIX="$PREFIX/llvm-$LLVM_VERSION"
 LLVM_MAJOR_VERSION="${LLVM_VERSION%%\.*}"
 LLVM_MINOR_VERSION=${LLVM_VERSION#*\.}
 LLVM_MINOR_VERSION="${LLVM_MINOR_VERSION%\.*}"
-LLVM_CMAKE_CONFIG_DIR=share/llvm/cmake
-if [ $LLVM_MAJOR_VERSION -gt 3 ]; then
-	LLVM_CMAKE_CONFIG_DIR=lib/cmake/llvm
-elif [ $LLVM_MAJOR_VERSION -ge 3 -a $LLVM_MINOR_VERSION -ge 9 ]; then
-	LLVM_CMAKE_CONFIG_DIR=lib/cmake/llvm
-fi
```
Comment (mchalupa) [inline system-build.sh:242]: Would it be possible to preserve this part in the case that llvm-config is not found or is not provided?

Comment (lzaoral) [reply system-build.sh:242]: Sure, I can put it back. But if I'm not mistaken, even the old `system-build.sh` would just exit
if `llvm-config` is not found:
```shell
test -z "$LLVM_CONFIG" && LLVM_CONFIG=$(which llvm-config || true)

if [ ! -z $LLVM_CONFIG -a -x $LLVM_CONFIG ]; then
	echo "Using llvm-config: $LLVM_CONFIG";
else
	exitmsg "Cannot find llvm-config binary. Try using llvm-config= switch"
fi
```
I can look into it. Do you have any reproducible scenario where LLVM is present but not found?

**Comment by lzaoral** [comment]
I have just rebased the PR with master to fix the conflicts.

**Comment by lzaoral** [comment]
I have reworked the search for directory with `LLVMConfig.cmake` and simplified other commits, so that LLVM 3.9 and lower can be used with `system-build.sh` too (both system or provided). (Some parts of Symbiotic fail to compile with these, but that's a separate issue.)

File: system-build.sh (hunk 1 above)
Comment (mchalupa) [reply system-build.sh:247]: Oh, you're right.

**Comment by mchalupa** [comment]
Thank you for this PR.

**Pull request merged**